catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -52,95 +57,194 @@
 
   private static final String NOT_AVAILABLE = "N/A";
 
+  @Option(
+    name = "--input",
+    required = false,
+    aliases = {"-i", "-input"},
+    multiValued = false,
+    description = "Displays only input transformers"
+  )
+  private boolean inputOption = false;
+
+  @Option(
+    name = "--metacard",
+    required = false,
+    aliases = {"-m", "-metacard"},
+    multiValued = false,
+    description = "Displays only metacard transformers"
+  )
+  private boolean metacardOption = false;
+
+  @Option(
+    name = "--all",
+    required = false,
+    aliases = {"-a", "-all"},
+    multiValued = false,
+    description = "Displays all the properties of the desired transformers"
+  )
+  private boolean allOption = false;
+
   @Override
   protected Object executeWithSubject() throws Exception {
 
-    // ServiceReferences are needed to get property information from InputTransformers
-    List<ServiceReference<InputTransformer>> serviceReferences =
-        (List<ServiceReference<InputTransformer>>)
-            bundleContext.getServiceReferences(InputTransformer.class, "(id=*)");
+    if (inputOption) {
+      printTransformers(InputTransformer.class, INPUT);
+      return null;
+    }
+    if (metacardOption) {
+      printTransformers(MetacardTransformer.class, METACARD);
+      return null;
+    }
+
+    printTransformers(InputTransformer.class, INPUT);
```

Comment:
This logic looks a little weird to me. It looks like 
- `-i -m` will just print `InputTransformer`s
- neither `-i` nor `-m` will print both `InputTransformer`s and `MetacardTransformer`s

Is that intended?

Maybe
```java
  @Override
  protected Object executeWithSubject() throws Exception {
    if (!inputOption && !metacardOption) {
      // print some message about expecting the `--input` and/or the `--metacard` options
    } else {
        if (inputOption) {
          printTransformers(InputTransformer.class, INPUT);
        }
        if (metacardOption) {
          printTransformers(MetacardTransformer.class, METACARD);
        }
    }

    return null;
  }
```
or
```java
  @Override
  protected Object executeWithSubject() throws Exception {
    if (!inputOption && !metacardOption) {
      inputOption = true;
      metacardOption = true;
    }

    if (inputOption) {
      printTransformers(InputTransformer.class, INPUT);
    }
    if (metacardOption) {
      printTransformers(MetacardTransformer.class, METACARD);
    }

    return null;
  }
```
would work better.

Reply:
oops. I forgot about the case with both options. will fix